File: web_servers/rs_hyper/src/usecases/users/entities.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct User {
    pub user_id: i32,
    pub username: String,
    pub email: Option<String>,
    pub is_archived: bool,
    pub created_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct UserForUpdate {
    pub user_id: i32,
    pub username: String,
    pub email: String,
}

impl User {
    pub fn new(
        user_id: i32,
        username: String,
        email: Option<String>,
        is_archived: bool,
        created_at: DateTime<Utc>,
    ) -> User {
        User {
            user_id,
            username,
            email,
            is_archived,
            created_at: created_at.to_rfc3339(),
        }
    }
}

File: web_servers/py_fast_api_uvicorn/src/db/entities.py
import datetime
import ujson
from pydantic import BaseModel


class User(BaseModel):
    user_id: int
    username: str
    email: str | None
    created_at: datetime.datetime
    updated_at: datetime.datetime
    is_archived: bool

class UpdateUser(BaseModel):
    username: str
    email: str | None
